docs/reference/esql/time-spans.asciidoc

```diff
@@ -0,0 +1,106 @@
+[[esql-time-spans]]
+=== {esql} time spans
+
+++++
+<titleabbrev>Time spans</titleabbrev>
+++++
+
+Time spans provide a way of describing a specific period of time that separates two datetime values. There are currently two supported types of time spans, `DATE_PERIOD` specifies intervals in years, quarters, months, weeks and days, and `TIME_DURATION` specifies intervals in hours, minutes, seconds and milliseconds.
+
+The time span contains two elements, an integer value and a temporal unit, both of them are required when specifying a time span.
+
+Time spans are often used in datetime related grouping functions like <<esql-bucket, BUCKET>>, scalar functions like <<esql-date_trunc, DATE_TRUNC>> and arithmetic operators like `+` and `-`. There are conversions functions that can convert string literals to time spans, <<esql-to_dateperiod, TO_DATEPERIOD>>, <<esql-to_timeduration, TO_TIMEDURATION>> and their corresponding cast operators `::DATE_PERIOD` and `::TIME_DURATION`.
```

Comment:
For `+` and `-` we could also link to the [operators page](https://www.elastic.co/guide/en/elasticsearch/reference/master/esql-functions-operators.html), since these list the time span types as supported, though I don't know how to add the slug (`#_add`) -- maybe @leemthompo knows?

Reply:
`cast operators` could also be [linked](https://www.elastic.co/guide/en/elasticsearch/reference/master/esql-functions-operators.html#esql-cast-operator).

Reply:
> For `+` and `-` we could also link to the [operators page](https://www.elastic.co/guide/en/elasticsearch/reference/master/esql-functions-operators.html), since these list the time span types as supported, though I don't know how to add the slug (`#_add`) -- maybe @leemthompo knows?

I figured out a way to link them!